airgun/entities/subscription.py

```diff
@@ -0,0 +1,211 @@
+        )
+        wait_for(
+                lambda: self.has_manifest == has_manifest,
+                handle_exception=True, timeout=10,
+                logger=view.logger
+        )
+        view.flash.dismiss()
+
+    @property
+    def has_manifest(self):
+        """Is there manifest present in current organization?
+        :return: boolean value indicating whether manifest is present
+        """
+        view = self.navigate_to(self, 'All')
+        return not view.add_button.disabled
+
+    def add_manifest(self, manifest_file):
+        """Upload manifest file
+        :param manifest_file: Path to manifest file
+        """
+        view = self.navigate_to(self, 'Manage Manifest')
+        view.wait_animation_end()
+        view.fill({
+            'manifest.manifest_file': manifest_file,
+        })
+        self._wait_for_process_to_finish('Import Manifest', has_manifest=True)
+
+    def delete_manifest(self):
+        """Delete manifest from current organization"""
+        view = self.navigate_to(self, 'Delete Manifest Confirmation')
+        view.wait_animation_end()
+        view.delete_button.click()
+        self._wait_for_process_to_finish('Delete Manifest', has_manifest=False)
+
+    def read_delete_manifest_message(self):
+        """Read message displayed on 'Confirm delete manifest' dialog"""
+        view = self.navigate_to(self, 'Delete Manifest Confirmation')
+        view.wait_animation_end()
+        return view.message.read()
+
+    def add(self, entity_name, quantity=1):
+        """Attach new subscriptions
+        :param entity_name: Name of subscription to attach
+        :param quantity: Number of subscriptions to attach
+        """
+        view = self.navigate_to(self, 'Add')
+        for row in view.table.rows(subscription_name=entity_name):
+            row['Quantity to Allocate'].fill(quantity)
+        view.submit_button.click()
+        self._wait_for_process_to_finish(
+                'Bind entitlements to an allocation', has_manifest=True)
+
+    def search(self, value):
+        """search for subscription"""
+        view = self.navigate_to(self, 'All')
+        return view.search(value)
+
+    def provided_products(self, entity_name):
+        """Read list of products provided by subscription
+        :param entity_name: Name of subscription
+        :return: List of strings with product names
+        """
+        view = self.navigate_to(self, 'Details', entity_name=entity_name)
+        return view.details.provided_products.read()
+
+    def enabled_products(self, entity_name):
+        """Read list of enabled products provided by subscription
+        Catches possible exception to always return known data structure
+        :param entity_name: Name of subscription
+        :return: List of strings with product names (may be empty)
+        """
+        view = self.navigate_to(self, 'Details', entity_name=entity_name)
+        view.enabled_products.select()
```

Comment:
nitpick: not needed anymore. Next line you access `enabled_products` tab widget, and when widget from specific tab is access, it gets selected automatically.